### ISSUE TITLE
Lubuntu default logout dialog

### ISSUE THREAD

**Issue by antis81** [lxde/lxqt#579]
With the current development version of LXQt 0.9+ (and Lubuntu 14.10) it is apparently still not possible to leave an LXQt session via `lxsession-logout`. Unfortunately I cannot check `lubuntu-logout`, which is part of the APT package `lubuntu-default-settings`. When trying to install it, APT wants to remove `lxqt-common`.

Why do I need that? Well, I'm using a panel replacement (cairo/glx-dock) and the most simple (and best working) way to get a logout dialog is by calling `lubuntu-logout` - as stated above that's not available. So I tried `lxsession-logout`. But this only seems to quit OpenBox. I.e. window borders disappear, but the session stays intact. This behaviour is the same in a plain Lubuntu 14.10 installation (without LXQt installed). 

Some time ago there was `lxqt-logout`, which worked well, but now got removed ....
:pencil2: EDIT: It's called `lxqt-power` - my mistake, sorry.

In short: How can I teach LXQt 0.9 to properly run the default Lubuntu logout dialog?

**Comment by Vladimir-csp** [comment]
I don`t know the answer to your question, but "leave actions" in LXQt are now presented via simple .desktop files. You can add them to cairo dock as applications.

**Comment by MttCastelli** [comment]
@antis81 lxsession is part of lxde,  but no more used by lxqt: you just have to install 'lxqt-metapackage', and you can get rid of a lot of lubuntu packages, if you want. The new current 'lxqt-leave' command is in 'lxqt-session' package. 
(I use lubuntu 15.04 + lubuntu-dev PPA + lxqt-metapackage, just to let you know.)

**Comment by antis81** [comment]
@MttCastelli, @Vladimir-csp Thanks for your answers. What I'm looking for is a replacement for `lxqt-power`. I.e. a "classic" shutdown dialog.

This dialog should be callable by a trigger button (yes, there are also two keyboards with a real power button on my desk :smile:).

FYI: Cairo-Dock internally generates .desktop files. What I do as a workaround (runs for several months now): I added a custom "shutdown" applet, that calls `lxqt-leave --logout`.

Had a short peek into the `lxqt-leave` code. It simply wraps calls to `powermanager`. So the question is: Is there a GUI? If no, why not? Why got `lxqt-power` removed without a proper replacement?

**Comment by antis81** [comment]
For the record, I already wrote some lines about this in #425 and followed the discussion as good as possible. This issue is still unresolved. Apparently `lxqt-power` got dropped for good in favour to `lxqt-leave`. There's no problem with that decision. It's just that there is not dialog available. In addition, the distro's default logout dialog will not work with LXQt.

@surlykke Could you please help on sorting this out?

**Comment by pmattern** [comment]
I'd just like to mention briefly that something roughly similar is addressed in #477.

**Comment by antis81** [comment]
@pmattern Thanks.

**Comment by paulolieuthier** [comment]
@antis81 lxqt-session provides .desktop files for power actions. `lxqt-hibernate.desktop`, lxqt-lockscreen.desktop`, `lxqt-logout.desktop`, `lxqt-reboot.desktop`, `lxqt-shutdown.desktop` and `lxqt-suspend.desktop`.

If you run one or those, you'll get a confirmation dialog (if you can disable the dialog in lxqt-config-powermanagement). What else lxqt-power provided that is not provided by lxqt-leave?

**Comment by antis81** [comment]
@paulolieuthier Thanks for answering.
>If you run one or those, you'll get a confirmation dialog (if you can disable the dialog in lxqt-config-powermanagement). What else lxqt-power provided that is not provided by lxqt-leave?

Just to set it right: The option to enable/disable the confrmation dialog is in lxqt-config-session (which totally makes sense :smile:), not lxqt-config-powermanagement. But this is not what I'm talking about.

Let me answer your latter question by providing a visual sample of my customized desktop with the missing dialog. As I didn't find a screenshot of the original lxqt-power on the web, I created (an ugly) one out of the ui file's preview. Anyways ... here it is:

![logout-dialog-sample](https://cloud.githubusercontent.com/assets/440517/7158442/6bce3614-e37a-11e4-9543-52dbcad31e79.png)

The dialog should appear when ...
1. ... clicking the "power button" on the top-right screen corner
2. ... the power button on a keyboard (macbook/external iMac G3 keyboard in my case)
3. ... called by terminal (e.g. `lxqt-leave --gui` or `lxqt-session-logout` ...)

The dilemma lies currently in the above mentioned package clash (`lubuntu-default-settings` vs. `lxqt-common`). This makes it impossible to use the distro's default logout dialog.

IMHO this dialog should be part of the LXQt desktop environment (i.e. `lxsession-logout` => `lxqt-session-logout` or `lxqt-session-leave`). When it comes to Lubuntu, it wraps `lxsession-logout` in `lubuntu-logout`.

**Comment by paulolieuthier** [comment]
> Just to set it right: The option to enable/disable the confrmation dialog is in lxqt-config-session 
(which totally makes sense :smile:), not lxqt-config-powermanagement.

Oh, right. Sorry. I, for one, should know it for I implemented it :smile:.

I totally get it now. There should be a gui for all the power actions. I think that could go in `lxqt-leave`. I'll work on it.

**Comment by antis81** [comment]
Nothing to add - Thanks! :+1: :+1: :+1:

Btw.: I'd really like to help out and taking part in the project. Be it with coding, translation, webpage, docs, .... Just contact me and I'll do my very best.

**Comment by paulolieuthier** [comment]
@antis81 that's very good news. IMO, the best way to help besides coding is testing and filing issues. About coding, please take a look at the issue tracker and start playing with a lxqt component. Read more [here](https://github.com/lxde/lxqt/wiki/Contributing-code).

**Comment by paulolieuthier** [comment]
Sorry.

**Comment by paulolieuthier** [comment]
@antis81 please update lxqt-session and try `lxqt-leave --gui`.

**Comment by antis81** [comment]
@paulolieuthier You mean like update and install from source? I'm asking as the change is not in the upstream PPA yet.

**Comment by antis81** [comment]
For dev. environment I cloned the lxqt repos and created a `build` directory in `lxqt/lxqt-session`. Following the build instructions, I run the following command (some personal extensions required): `PATH=$HOME/Qt/5.4/gcc_64/bin:$PATH cmake -DQT5=ON -- ..`

I have two questions about compiling:
1. CMake errors out and keeps telling me to install `KF5WindowSystem`. How can I do that in a minimal way? I couldn't find any instructions on that so far.
2. How do I enable C++11 features (I have those available)?

**Comment by palinek** [comment]
> 1. CMake errors out and keeps telling me to install KF5WindowSystem. How can I do that in a minimal way? I couldn't find any instructions on that so far.

In debian/ubuntu the dev package for `KF5WindowSystem` is: `libkf5windowsystem-dev`. For all future build dependencies that cmake is complaining you have to find the package that contains `XXX.cmake` (see how cmake `find_package` works for example here: http://www.cmake.org/cmake/help/v3.0/command/find_package.html)

> 2. How do I enable C++11 features (I have those available)?

You can see it in other lxqt modules. Here http://stackoverflow.com/questions/10851247/how-to-activate-c-11-in-cmake is some discussion also

**Comment by antis81** [comment]
@palinek Thanks for the info. The following should do the trick:

`sudo apt-get install liblxqt-dev liblxqt5-dev'
(I installed libkf5windowsystem-dev separately, which should not be necessary - I'll recheck that).

After doing so, lxqt-session compiles fine with Qt5.4.1 (by adding the PATH prefix).

@paulolieuthier Dialog runs fine and basically works. However, there's always some small things to make it better :smile::
* Positioning: The dialog saves it's last position. It should at least appear on the screen with the mouse focus (as I configured here by openbox). Maybe also centered (like lxqt-runner). 
* Look & Feel: I really like the transparent and "icons only" layout (with tooltips) from the original lxqt-power dialog. If I find a screenshot, I'll post it here. Just makes it look more "cleaned up" :smile:.
* Add a .desktop file, representing the action (e.g. to provide an "applet" in lxqt-panel or elsewhere).
* Dialog is quit when returning from suspend/hibernate mode. Actually this behaviour makes sense.  though it "feels a bit weird" as I'd expect the exact (!) same state as before suspending.
* Last not least: i18n translation of course.

Here's how it appears on my Desktop:
![lxqt-leave-dialog](https://cloud.githubusercontent.com/assets/440517/7179392/a20cc642-e437-11e4-8748-e991f83bb157.png)

Many thanks and kudos to you, finally getting this in :+1:.

**Comment by paulolieuthier** [comment]
@antis81 I fixed the position and created the `.desktop` file. The other issues may be fixed in the future. Translation will certainly come.
